### ISSUE TITLE
Add positionat and velocityat counterparts for orbit objects

### ISSUE THREAD

**Issue by pand5461** [KSP-KOS/KOS#2529]
Sometimes it's useful to have the ability to get the predicted state on a specific orbit patch, instead of getting state for a specific object.

An example: after a maneuver node is executed, compute how far from the planned trajectory the ship is going to be in X time and possibly make corrections until the difference vanishes.

**Comment by pand5461** [comment]
A similar problem has been mentioned in  #1024.

**Comment by Tricky1702** [comment]
Agreed. I use a mathematical equation to predict the future apoapsis for the next burn for Hohmann transfers. Based on arc calculations. Not sure how accurate it is though.

**Comment by mdpkh** [comment]
Still wanting this feature. I'm trying to build a constellation of relays, each in a circular orbit but in different planes and with their positions otherwise coordinated. When I launch a new satellite, I can't see its target orbit in map view because there's no object on that orbit (yet), so planning maneuvers manually is difficult and very imprecise. However, I can create an `Orbit` object by copying the orbital elements of the first relay and modifying them in a prescribed way. If I could query the position of a hypothetical object on that orbit at a given point in time, I could plan maneuvers to bring my craft to that point at that time, and then if I could query the velocity of the hypothetical object at that time, I could plan a final maneuver to make my craft's velocity match it, and then I'd have a precise plan to place my craft on the prescribed orbit.

The position and velocity of a hypothetical object on a given orbit at a given time is exactly what `Orbit:POSITIONAT()` and `Orbit:VELOCITYAT()` should do. No need to worry about maneuvers or patches or actual `Orbitable` objects, as that's what the suffixes of `Orbitable` objects already do. To more thoroughly describe the relationship: `Orbitable:POSITIONAT()` and `Orbit:VELOCITYAT()` can be described as returning the `Orbit:POSITIONAT()` and `Orbit:VELOCITYAT()` values of the patch that the `Orbitable` is predicted to be on at the given time.

**Comment by mdpkh** [comment]
It looks to me like the member functions `GetPositionAtUT` and `GetVelocityAtUT` of the class `OrbitInfo` already exist and do exactly what I want; they just need to be bound to suffixes. I'd do it myself if I were equipped to compile KSP/Unity mods.

On the other hand, I suppose I have a reasonable alternative approach to the use case described above: I can create and manipulate maneuver nodes until the orbital elements of the last patch match the orbital elements of the desired target orbit to within some tolerance.

**Comment by nuggreat** [comment]
There is also [this posted](https://www.reddit.com/r/Kos/comments/6bsg5g/velocity_vectors_along_arbitrary_orbits/) over on the kOS subreddit that can calculate the velocity vector for an arbitrary orbit.  A little bit of work and you can also get the position as well.
